### ISSUE TITLE
Incorrect computation of gradients for transposed 2d convolution when stride=2 on CPU

### ISSUE THREAD

**Issue by artyom-beilis** [pytorch/pytorch#63314]
## Bug

Incorrect backpropogation calculation for stride=2 transpose2d

## To Reproduce

    import torch
    torch.manual_seed(100)
    conv=torch.nn.ConvTranspose2d(1,1,kernel_size=1,bias=False,stride=2)
    x=torch.randn(1,1,1,2,requires_grad=True)
    y=conv(x)
    yd=torch.tensor([[[[1.5,0,0.7]]]])
    y.backward(yd,retain_graph=True)
    print("W:",conv.weight)
    print("X:",x)
    print("Y:",y)
    print("dY:",yd)
    print("dX:",x.grad)

Output

    W: Parameter containing:
    tensor([[[[-0.7767]]]], requires_grad=True)
    X: tensor([[[[-1.9954, -0.3385]]]], requires_grad=True)
    Y: tensor([[[[1.5498, 0.0000, 0.2629]]]], grad_fn=<SlowConvTranspose2DBackward>)
    dY: tensor([[[[1.5000, 0.0000, 0.7000]]]])
    dX: tensor([[[[-1.1650,  0.0000]]]])


## Expected behavior

Expecting: dx[0,0,0,1] = 0.7 * -0.7767 instead of 0. The 2nd dY is taken instead of thirdt

Why - the X affects Y at place 2  `0.2629 = -0.7767 * -0.3385` correctly but in backward dX takes dY at [0,0,0,1] instead of dy at [0,0,0,2]

Note if I run  this on cuda I get correct result

    import torch
    torch.manual_seed(100)
    conv=torch.nn.ConvTranspose2d(1,1,kernel_size=1,bias=False,stride=2).to('cuda')
    x=torch.randn(1,1,1,2).to('cuda')
    x.requires_grad=True
    y=conv(x)
    yd=torch.tensor([[[[1.5,0,0.7]]]],device='cuda')
    y.backward(yd,retain_graph=True)

    print("W:",conv.weight)
    print("X:",x)
    print("Y:",y)
    print("dY:",yd)
    print("dX:",x.grad)

Last line is:

    dX: tensor([[[[-1.1650, -0.5437]]]], device='cuda:0')

As expected


## Environment

Collecting environment information...
PyTorch version: 1.8.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04 LTS (x86_64)
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04) 7.4.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.25

Python version: 3.6.9 (default, Jan 26 2021, 15:33:00)  [GCC 8.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-50-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Quadro P3200 with Max-Q Design
Nvidia driver version: 460.73.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A                                                                                                                                                                  
                                                                                                                                                                                             
Versions of relevant libraries:                                                                                                                                                              
[pip3] numpy==1.19.5                                                                                                                                                                         
[pip3] torch==1.8.1                                                                                                                                                                          
[pip3] torchvision==0.9.1                                                                                                                                                                    
[conda] Could not collect



cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @VitalyFedyunin @jamesr66a @ppwwyyxx

**Comment by ngimel** [comment]
Can't repro on master, this was likely fixed by mkldnn update.

**Comment by artyom-beilis** [comment]
I see it is indeed fixed in nightly
